src/rpc/mempool.cpp

```diff
@@ -669,6 +670,133 @@ static RPCHelpMan savemempool()
+                        {RPCResult::Type::STR_HEX, "txid", "The transaction hash in hex"},
+                        {RPCResult::Type::STR_HEX, "wtxid", "The transaction witness hash in hex"},
+                        {RPCResult::Type::NUM, "vsize", "Virtual transaction size as defined in BIP 141."},
+                        {RPCResult::Type::OBJ, "fees", "Transaction fees (only present if result=\"valid\")", {
+                            {RPCResult::Type::STR_AMOUNT, "base", "transaction fee in " + CURRENCY_UNIT},
+                        }},
+                    }}
+                }},
+                {RPCResult::Type::STR_AMOUNT, "package-feerate", /*optional=*/ true, "package feerate in " + CURRENCY_UNIT + " per KvB. Not always available."},
+                {RPCResult::Type::ARR, "replaced-transactions", /*optional=*/ true, "List of txids of replaced transactions",
+                {
+                    {RPCResult::Type::STR_HEX, "", "The transaction id"},
+                }},
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("testmempoolaccept", "[rawtx1, rawtx2]") +
+            HelpExampleCli("submitrawtransaction", "[rawtx1, rawtx2]")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    if (!Params().IsMockableChain()) {
+        throw std::runtime_error("submitrawpackage is for regression testing (-regtest mode) only");
+    }
+    RPCTypeCheck(request.params, {
+        UniValue::VARR,
+    });
+    const UniValue raw_transactions = request.params[0].get_array();
+    if (raw_transactions.size() < 1 || raw_transactions.size() > MAX_PACKAGE_COUNT) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Array must contain between 1 and " + ToString(MAX_PACKAGE_COUNT) + " transactions.");
+    }
+
+    std::vector<CTransactionRef> txns;
+    txns.reserve(raw_transactions.size());
+    for (const auto& rawtx : raw_transactions.getValues()) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, rawtx.get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR,
+                               "TX decode failed: " + rawtx.get_str() + " Make sure the tx has at least one input.");
+        }
+        txns.emplace_back(MakeTransactionRef(std::move(mtx)));
+    }
+
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    CTxMemPool& mempool = EnsureMemPool(node);
+    CChainState& chainstate = EnsureChainman(node).ActiveChainstate();
+    const auto package_result = WITH_LOCK(::cs_main, return ProcessNewPackage(chainstate, mempool, txns, /* test_accept */ false));
+
+    // First catch any errors.
+    switch(package_result.m_state.GetResult()) {
+        case PackageValidationResult::PCKG_RESULT_UNSET: break;
+        case PackageValidationResult::PCKG_POLICY:
+        case PackageValidationResult::PCKG_MEMPOOL_ERROR:
+        {
+            throw JSONRPCTransactionError(TransactionError::INVALID_PACKAGE,
```

Comment:
Well sounds `PackageValidationResult::PCKG_MEMPOOL_ERROR` has its equivalent `TransactionError::MEMPOOL_ERROR` ? Or other reason to use `INVALID_PACKAGE` ?

Reply:
Good point, `TransactionError::MEMPOOL_ERROR` is more appropriate here. Switched.

Reply:
Good catch, swapped